### ISSUE TITLE
 Value cannot be null. (Parameter 'jsRuntime')

### ISSUE THREAD

**Issue by nssidhu** [IvanJosipovic/BlazorApplicationInsights#107]
I am using .NET RC1 Blazor WebAssembly hosted app

Getting the following Error.

crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Value cannot be null. (Parameter 'jsRuntime')
System.ArgumentNullException: Value cannot be null. (Parameter 'jsRuntime')
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at BlazorApplicationInsights.ApplicationInsights.TrackEvent(String name, Dictionary`2 properties)
   at GetInLineV6New.Client.Pages.Index.OnInitializedAsync() in C:\Users\admin\source\repos\GetInLineV6New\GetInLineV6New\Client\Pages\Index.razor.cs:line 25
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()

Index.razor
```
protected override async Task OnInitializedAsync()
    {
       
        await AppInsights.TrackEvent("My Event from Index Page");  //Line 25
}
```

**Comment by nssidhu** [comment]
you are right

**Comment by nssidhu** [comment]
Still getting that error

In Index.razor

```
protected override async Task OnAfterRenderAsync(bool firstRender)
    {
       if(firstRender)
        {
            await Task.Delay(200);
            await AppInsights.TrackEvent("My Event from Index Page");
        }
     
        await base.OnAfterRenderAsync(firstRender);
        return;
    }


```

Here is the Error

```
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not find 'appInsights.trackEvent' ('appInsights' was undefined).
      Error: Could not find 'appInsights.trackEvent' ('appInsights' was undefined).
          at https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:328
          at Array.forEach (<anonymous>)
          at a.findFunction (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:296)
          at _ (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:2437)
          at https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:3290
          at new Promise (<anonymous>)
          at Object.beginInvokeJSFromDotNet (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:3271)
          at Object.At [as invokeJSFromDotNet] (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:58349)
          at _mono_wasm_invoke_js_blazor (https://XXXXXXXX6new.azurewebsites.net/_framework/dotnet.6.0.0-rc.1.21451.13.js:1:193753)
          at wasm://wasm/00945eda:wasm-function[2590]:0x8073e
Microsoft.JSInterop.JSException: Could not find 'appInsights.trackEvent' ('appInsights' was undefined).
Error: Could not find 'appInsights.trackEvent' ('appInsights' was undefined).
    at https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:328
    at Array.forEach (<anonymous>)
    at a.findFunction (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:296)
    at _ (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:2437)
    at https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:3290
    at new Promise (<anonymous>)
    at Object.beginInvokeJSFromDotNet (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:3271)
    at Object.At [as invokeJSFromDotNet] (https://XXXXXXXX6new.azurewebsites.net/_framework/blazor.webassembly.js:1:58349)
    at _mono_wasm_invoke_js_blazor (https://XXXXXXXX6new.azurewebsites.net/_framework/dotnet.6.0.0-rc.1.21451.13.js:1:193753)
    at wasm://wasm/00945eda:wasm-function[2590]:0x8073e
   at Microsoft.JSInterop.JSRuntime.<InvokeAsync>d__16`1[[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime , String , Object[] )
   at BlazorApplicationInsights.ApplicationInsights.TrackEvent(String name, Dictionary`2 properties)
   at XXXXXXXX6new.Client.Pages.Index.OnAfterRenderAsync(Boolean firstRender)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task , ComponentState )
```

**Comment by mlienau** [comment]
I also ran into this issue. The fix for me was that `@using BlazorApplicationInsights` is **required** in your `_Imports.cshtml` file, even though you inject `IApplicationInsights` which comes from the `BlazorApplicationInsights.Interfaces` namespace. I had assumed that the interface had been moved at one point, and the readme wasn't updated. That was a bad assumption on my part, and I would like to know the actual problem. I ended up adding `BlazorApplicationInsights.Interfaces` to my imports file as well.
